docs/contrib-guide.md

```diff
@@ -0,0 +1,158 @@
+# Contribution Guide
+
+1. [Prerequisites](#prerequisites)
+2. [Build](#build)
+3. [Check](#check)
+4. [Testing](#testing)
+5. [Pull-request](#pull-request)
+6. [License](#license)
+
+Thank you for your time and effort to help us improve samba-operator.
+Here are a few steps to get started, with reference to some useful
+resources.
+
+
+## Prerequisites
+
+Development effort takes place using Linux environment and requires at
+minimum:
+
+1. [Go 1.18](https://golang.org/dl/) installed
+2. [GitHub](https://github.com/) account
+3. Development tools: git and make
+4. Testing: [minikube](https://minikube.sigs.k8s.io)
+
+Development collaboration takes place via github's samba-in-kuberbetus
+([SINK](https://github.com/samba-in-kubernetes/)) pages and facilities.
+
+The top-level [Makefile](../Makefile) is the entry point for various
+build commands. Few utilities are required during the build process,
+either for bootstrapping the project, building it or checking the
+validity of the source code itself. If not present in your `$PATH`,
+those tools may be installed locally to the project (under the
+local `.bin` directory) using:
+
+```sh
+  $ make build-tools
+  ...
+  $ ls .bin/
+```
+
+## Build
+
+The following section describes how to build and test samba-operator.
+A more detailed description of this process may be found in the
+[developer-notes](./developer-notes.md) document.
+
+Building the samba-operator is straightforward action with `make build`.
+Upon successful build, the output binary is written to `bin/manager`:
+
+```sh
+  $ make build
+  ...
+  $ ls bin/
+  manager
+```
+
+A developer may override some of the default Makefile settings via
+the `devel.mk` file at the project's root directory:
+
+```sh
+  $ cp devel.mk.sample devel.mk
+  $ vim devel.mk
+```
+
+## Check
+
+Before submitting any pull request, a developer must ensure that
+his code satisfy the following minimum code-style requirements:
+
+1. Go code conform to the project's standards:
+```sh
+  $ make check-format
+  $ make check-golangci-lint
+  $ make check-revive
+  $ make check-gosec
+```
+
+2. YAML files conform to the project's standards:
+```sh
+  $ make check-yaml
+```
+
+3. Unit-tests pass without errors:
+```sh
+  $ make test
+```
+
+
+The above steps may also be executed using single make rule:
+`make check test`. If any of those steps fail they must first be
+fixed before any other developer will review your patches.
+
+## Testing
+
+Before submitting any pull request, a developer should test his code
+on a kubernetes cluster using samba-operator built-in test suite,
+typically, using [minikube](https://minikube.sigs.k8s.io) and a
+private image:
```

Comment:
Even running minikube can be a PITA. I would recommend but not require this, especially in a word where draft PRs can be created. So:
```suggestion
Ideally, before submitting any pull request, a developer will test any code changes locally.
Local clusters are simpler to debug and, if one has not had code merged to the project before,
it does not require approval from the project maintainers. A prerequisite to running the
samba-operator test suite is a running Kubernetes cluster. Typically this will be [minikube](https://minikube.sigs.k8s.io) but other Kubernetes implementations can suffice.
An example using minikube and a private container image follows:
```

Reply:
OK. Here I wanted also to emphasize to the naive developer that there is more than `make test` in this project. That said, I accept your suggestion for re-phrasing.